presto-testing/src/main/java/io/prestosql/testing/AbstractTestJoinQueries.java

```diff
@@ -2275,6 +2275,39 @@ public void testEquijoinOnDifferentTypesWithFilter()
                 "WHERE id3 = 10");
     }
 
+    @Test
+    public void testMultiJoinWithEligibleForDynamicFiltering()
```

Comment:
I think you can synthetize test case without DF, e.g simiar to:
```
   B (x=y)
  /      \
 /         \
A (x=3)     \
            outer join C (I'm not sure what have to be here)
```

Reply:
```
SELECT *
FROM (SELECT * FROM region WHERE regionkey = 3) r -- A
JOIN (
    SELECT *
    FROM (SELECT * FROM nation WHERE nationkey = 3) n -- D
    LEFT JOIN (SELECT * FROM customer) c  -- E
   ON n.nationkey = c.custkey -- C
) ON r.regionkey = c.custkey -- B
```
?

-- this works today

Reply:
`-- this works today`

Most likely because it get converted to INNER join at first PPD run, when join distribution is not set yet